lambda_src/drivers/destination_s3.py

```diff
@@ -89,25 +87,37 @@ def initialize(destination: Text, batch_id: Text):
     # Regex captures characters after and including the rightmost '/' in a path,
     # which are then replaced with a '/', e.g. '/a/b/c' -> '/a/b/'
     prefix_folder = re.sub(r'/[^/]*$', '/', prefix) if '/' in prefix else ''
-
+    prefix_folder = re.sub(r'/{[^}].*', '/', prefix_folder)
     if prefix_folder:
         write_to_s3(bucket, prefix_folder, content)
 
 
 def write(
     destination: Text,
     batch_id: Text,
-    datum: Union[Dict, List],
+    datum: Union[Dict, List, bytes],
     row_index: int,
 ) -> Dict[Text, Any]:
     bucket, prefix = parse_destination_uri(destination)
     encoded_datum = (
-        '\n'.join(json.dumps(d) for d in datum)
+        datum
+        if isinstance(datum, bytes)
+        else ('\n'.join(json.dumps(d) for d in datum)).encode()
```

Comment:
bytes() needs an encoding like `utf-8`, when used with a string, so encoded both of the strings instead

Reply:
that's probably better, to avoid calling bytes on bytes.

the `.encode()` defaults to `utf-8`. an encoding is necessary because `bytes` are a specific abstraction while `str` is a generic with many encodings possible in memory/on disk/over network.

Reply:
Yess